### PR TITLE
add Bullet gem and Simplecov

### LINKED ISSUE #9: Add Bullet Gem and Simplecov
-  kill N+1 queries  and check test coverage

### LINKED ISSUE #11: Fix N+1 queries

### PR DESCRIPTION
**Pull request by snehithlal**
Fixes #9, Fixes #11 
- Added Bullet gem
- Added Simplecov gem
- Optimized query
- added gitignore for coverage
- updated .env for test database

### CODE REVIEW

**Comment by snehithlal** [comment]
@ahammednafih

File: Gemfile [hunk 1]
```diff
@@ -36,6 +36,9 @@ gem "jwt"
 gem "blueprinter"
 
 group :development, :test do
+  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
+  gem "bullet"
+  gem "byebug", platforms: %i[mri mingw x64_mingw]
```
Comment (ahammednafih) [inline Gemfile:41]: We already have pry, do we need byebug too?

File: db/schema.rb [hunk 2]
```diff
@@ -12,94 +12,95 @@
 
 ActiveRecord::Schema[7.0].define(version: 2022_01_13_124609) do
   # These are extensions that must be enabled in order to support this database
-  enable_extension "plpgsql"
+  enable_extension 'plpgsql'
```
Comment (ahammednafih) [inline db/schema.rb:15]: We do prefer double quotes for all now, updated rubocop. Please revert this back.

**Review by ahammednafih — CHANGES_REQUESTED** [review]
@snehithlal Added some suggestions.

**Comment by ahammednafih** [comment]
@snehithlal _a

File: Gemfile (hunk 1 above)
Comment (snehithlal) [reply Gemfile:41]: Came with the merge. will remove.

**Comment by snehithlal** [comment]
@ahammednafih updated the changes please review

File: Gemfile [hunk 3]
```diff
@@ -36,6 +36,8 @@ gem "jwt"
 gem "blueprinter"
 
 group :development, :test do
+  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
```
Comment (ahammednafih) [inline Gemfile:39]: Remove unnecessary comments.

File: Gemfile.lock [hunk 4]
```diff
@@ -252,6 +260,8 @@ DEPENDENCIES
   bcrypt (~> 3.1.7)
   blueprinter
   bootsnap (>= 1.4.4)
+  bullet
+  byebug
```
Comment (ahammednafih) [inline Gemfile.lock:264]: Update lock file

**Review by ahammednafih — CHANGES_REQUESTED** [review]

**Comment by ahammednafih** [comment]
@snehithlal please check the feedback and you can merge.

**Comment by snehithlal** [comment]
@ahammednafih  added README

**Pull request merged**